doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,432 @@
+
+The other important issue is decoding/encoding context access, which is described in :ref:`migration_tinycbor_to_zcbor_mcumgr_ctxt`.
+
+.. note::
+
+   The encoding context given to a command processing handler function has the top level map already created.
+   It will have the map closed on successful return from the handler, so handlers need not create the top level map.
+   In contrary to the above, the decoding context is not yet opened and handlers are responsible for opening the top level map before decoding of a command.
+
+.. _migration_tinycbor_to_zcbor_mcumgr_ctxt:
+
+Decoding and encoding context
+-----------------------------
+
+To command the processing of functions in TinyCBOR and zcbor implementation, the MCUmgr provides the following:
+
+ * A pointer to an object of type `struct mgmt_ctxt` from which the `encoder` element is used by CBOR encoding functions to write data.
+ * `it` - a  TinyCBOR `CborValue iterator to decoding functions.
+
+zcbor uses `zcbor_state_t` type objects for decoding and encoding states and these state objects have replaced the TinyCBOR `it` iterator and `encoder`, respectively, within `struct mgmt_ctxt`.
+The `zcbor_state_t` is not directly placed in `struct mgmt_ctxt` but is embedded within the `struct cbor_nb_reader` type object for decoding context, and in the `struct cbor_nb_writer` type object for encoding context.
+The `struct cbor_nb_reader` and `struct cbor_nb_writer` objects tie `zcbor_state_t` with `net_buf` type buffers that hold data for decoding or will hold encoded data.
+
+The replacement of decoder and encoder objects, as accessed through `struct mgmt_ctxt`, is shown by following table:
+
+.. table::
+    :align: center
+
+    +===================+===================+
+    | TinyCBOR          | zcbor             |
+    +-------------------+-------------------+
+    | `encoder`         | `cnbe.ts`         |
+    |                   | `cnbe->ts`        |
+    +-------------------+-------------------+
+    | `it`              | `cnde.ts`         |
+    |                   | `cnde->ts`        |
+    +---------------------------------------+
+
+.. _migration_tinycbor_to_zcbor_mcumgr_cborattr:
+
+Replacing `cbor_read_object` with `zcbor_map_decode_bulk`
+---------------------------------------------------------
+
+.. note::
+
+   In some scenarios, decoding of keys is not required or it is required for only a single key.
+   In such case, the procedure described in this section can be greatly simplified as a single key can be obtained by using `zcbor_tst_decode` looped until the key is found.
+
+`cborattr` is a private MCUmgr utility that is used within the command processing code to process CBOR list contents in bulk.
+It has been replaced with `zcbor_bulk`, also private to MCUmgr.
+Both APIs are centered around the main function `cbor_read_object` in case of TinyCBOR, and `zcbor_map_decode_bulk`, in case of zcbor.
```

Comment:
```suggestion
Above utility APIs are represented by a single function: `cbor_read_object` in case of TinyCBOR, and `zcbor_map_decode_bulk`, in case of zcbor.
```